src/test/kotlin/uk/gov/justice/digital/hmpps/makerecalldecisionapi/service/RecommendationServiceTest.kt

```diff
@@ -47,50 +52,56 @@ internal class RecommendationServiceTest : ServiceTestBase() {
-    assertThat(result.personOnProbation?.dateOfBirth).isEqualTo(LocalDate.parse("1982-10-24"))
-    assertThat(result.personOnProbation?.croNumber).isEqualTo("123456/04A")
-    assertThat(result.personOnProbation?.mostRecentPrisonerNumber).isEqualTo("G12345")
-    assertThat(result.personOnProbation?.nomsNumber).isEqualTo("A1234CR")
-    assertThat(result.personOnProbation?.pncNumber).isEqualTo("2004/0712343H")
-
-    // and
-    val captor = argumentCaptor<RecommendationEntity>()
-    then(recommendationRepository).should().save(captor.capture())
-    val recommendationEntity = captor.firstValue
-
-    assertThat(recommendationEntity.id).isNotNull()
-    assertThat(recommendationEntity.data.crn).isEqualTo(crn)
-    assertThat(recommendationEntity.data.status).isEqualTo(Status.DRAFT)
-    assertThat(recommendationEntity.data.personOnProbation).isEqualTo(PersonOnProbation(name = "John Smith", firstName = "John", middleNames = "Homer Bart", surname = "Smith", gender = "Male", ethnicity = "Ainu", dateOfBirth = LocalDate.parse("1982-10-24"), croNumber = "123456/04A", mostRecentPrisonerNumber = "G12345", nomsNumber = "A1234CR", pncNumber = "2004/0712343H"))
-    assertThat(recommendationEntity.data.lastModifiedBy).isEqualTo("Bill")
-    assertThat(recommendationEntity.data.lastModifiedDate).isEqualTo("2022-07-26T09:48:27.443Z")
-    assertThat(recommendationEntity.data.createdBy).isEqualTo("Bill")
-    assertThat(recommendationEntity.data.createdDate).isEqualTo("2022-07-26T09:48:27.443Z")
+      given(communityApiClient.getActiveConvictions(ArgumentMatchers.anyString()))
+        .willReturn(Mono.fromCallable { listOf(custodialConvictionResponse(sentenceDescription)) })
+
+      // and
+      given(recommendationRepository.save(any()))
+        .willReturn(recommendationToSave)
+
+      // when
+      recommendationService.createRecommendation(CreateRecommendationRequest(crn), "Bill")
+
+      // then
+      val captor = argumentCaptor<RecommendationEntity>()
+      then(recommendationRepository).should().save(captor.capture())
+      val recommendationEntity = captor.firstValue
+
+      assertThat(recommendationEntity.id).isNotNull()
+      assertThat(recommendationEntity.data.crn).isEqualTo(crn)
+      assertThat(recommendationEntity.data.status).isEqualTo(Status.DRAFT)
+      assertThat(recommendationEntity.data.personOnProbation).isEqualTo(PersonOnProbation(name = "John Smith", firstName = "John", middleNames = "Homer Bart", surname = "Smith", gender = "Male", ethnicity = "Ainu", dateOfBirth = LocalDate.parse("1982-10-24"), croNumber = "123456/04A", mostRecentPrisonerNumber = "G12345", nomsNumber = "A1234CR", pncNumber = "2004/0712343H"))
+      assertThat(recommendationEntity.data.convictionDetail).isEqualTo(
+        ConvictionDetail(
+          indexOffenceDescription = "Robbery (other than armed robbery)",
+          dateOfOriginalOffence = "2022-08-26",
+          dateOfSentence = "2022-04-26",
+          lengthOfSentence = "6 Days",
```

Comment:
if this gets displayed in the front end in future it would give more flexibility to send the number and the suffix in separate properties (is that how you're receiving them from delius?)

Reply:
I did think about this, and then though if it is ever displayed it will almost certainly be displayed together anyway i.e. it wouldn't make sense to show "6" or "Days" without the other. I can change if you think it is needed?

Reply:
Done